sys/reaper/reaper_unix.go

```diff
@@ -107,3 +108,59 @@ func (m *Monitor) Unsubscribe(c chan runc.Exit) {
 	close(c)
 	m.Unlock()
 }
+
+func (m *Monitor) getSubscribers() map[chan runc.Exit]struct{} {
+	out := make(map[chan runc.Exit]struct{})
+	m.Lock()
+	for k, v := range m.subscribers {
+		out[k] = v
+	}
+	m.Unlock()
+	return out
+}
+
+func (m *Monitor) notify(e runc.Exit) chan struct{} {
+	const timeout = 10 * time.Millisecond
+	var (
+		done    = make(chan struct{}, 1)
+		timer   = time.NewTimer(timeout)
+		success = make(map[chan runc.Exit]struct{})
+	)
+	stop(timer)
+
+	go func() {
+		defer close(done)
+
+		for i := 0; i < 100; i++ {
+			var (
+				failed      int
+				subscribers = m.getSubscribers()
```

Comment:
Is it expected to the get the whole list of subscribers in every loop vs having a static list?

Reply:
It could go both ways, we can always process the current list or process a static list from when the event happened